custom_components/tuya_local/devices/doel_tiplus_litterbox.yaml

```diff
@@ -0,0 +1,231 @@
+name: Litter box
+products:
+  - id: ijas23irjjaenllv
+    name: DOEL Ti+
+primary_entity:
+  entity: sensor
+  name: Cat weight
+  icon: "mdi:cat"
+  class: weight
+  dps:
+    - id: 6
+      type: integer
+      name: sensor
+      class: measurement
+      unit: kg
```

Comment:
I see you have added the unit back here, but the entity has a dp named "unit", which will always override it. Probably you had a reason to add this back in, but I think it points to a problem elsewhere in that case. If dp 108 is sometimes missing (which I suspect is your reason for adding static units), then it needs to be marked as optional to let the config be matched, and a null mapping for the default case should be added there to avoid the confusion over whether the static unit or the unit dp applies.